### PR TITLE
Chore: add jest and typescript

### PR DESCRIPTION
**Pull request by HeverFarber**
1. Add typescript to the project
2. Adjest "serverless-openapi-typescript" to types (all red lines)\
3. Add jest
4. Add all dev devDependencies
5. remove package-lock.json (npm i package-name don't take care of package-lock) 
6. Add github actions for CI

### CODE REVIEW

**Comment by HeverFarber** [comment]
[Add unit tests to documentation plugin](https://trello.com/c/553nx6vJ/4089-add-unit-tests-to-documentation-plugin)

File: src/serverless-openapi-typescript.ts [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
-const yaml = require('js-yaml');
-const fs = require('fs');
-
-class ServerlessOpenapiTypeScript {
-  constructor(serverless, options) {
+import type Serverless from "serverless";
+import fs from "fs";
+import yaml from "js-yaml";
+import tsj,  {SchemaGenerator} from "ts-json-schema-generator";
+import {upperFirst, mergeWith, set, isArray, get, isEmpty } from "lodash" ;
```
Comment (eyalatzmon) [inline src/serverless-openapi-typescript.ts:5]: would it be more clear to import lodash as 
```import _ from 'lodash';```?

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,6 @@ node_modules/
 .idea/
 .DS_Store
 .AppleDouble
-.LSOverride
\ No newline at end of file
+.LSOverride
+package-lock.json
```
Comment (RLRabinowitz) [inline .gitignore:6]: Why do we not want the lock?
AFAIK it's intended to exist in the git repo

**Review by RLRabinowitz — APPROVED** [review]
Looks good overall.

Left a question regarding package-lock

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# This workflow will do a clean install of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
```
Comment (roni-frantchi) [inline .github/workflows/ci.yml:3]:
```suggestion
```

File: .gitignore (hunk 2 above)
Comment (roni-frantchi) [reply .gitignore:6]: @HeverFarber you must have a lockfile or shrinkwrap to make the package consistently use the same dependency version on releases.

File: src/serverless-openapi-typescript.ts [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+  private readonly functionsMissingDocumentation: string[];
+  private readonly disable: boolean;
+  private hooks: { [hook: string]: () => {}};
+  private options: Options;
+  private typescriptApiModelPath: string;
+  private tsconfigPath: string;
+  private schemaGenerator: SchemaGenerator;
+
+  constructor(serverless: Serverless, options: Options) {
     this.serverless = serverless;
```
Comment (roni-frantchi) [inline src/serverless-openapi-typescript.ts:23]:
```suggestion
  private readonly functionsMissingDocumentation: string[];
  private readonly disable: boolean;
  private hooks: { [hook: string]: () => {}};
  private typescriptApiModelPath: string;
  private tsconfigPath: string;
  private schemaGenerator: SchemaGenerator;

  constructor(private serverless: Serverless, private options: Options) {
```

**Review by roni-frantchi — CHANGES_REQUESTED** [review]
@HeverFarber we must have a lockfile or shrinkwrap present on repo.  

Also I see that some style changes were introduced - consider adding prettier to avoid these

File: .gitignore (hunk 2 above)
Comment (HeverFarber) [reply .gitignore:6]: But this is not true for NPM package. it's true for CI. each CI manage his package-lock and his deployment.

But when you install package from package.json (npm i <package>) - the main project only use "dependencies" and install them and added them to the main package-lock. 

so if we will manage package-lock in this repo it only affects us for CI. and our package-lock can be different from what the user will install. and we will publish bad version.

do you still think we need package-lock?

File: src/serverless-openapi-typescript.ts (hunk 4 above)
Comment (HeverFarber) [reply src/serverless-openapi-typescript.ts:23]: I thought we don't like it. what about 

remove this line? 
```
this.serverless = serverless;
```

TS compile this line automatically when we add the private/public to the contractor

File: src/serverless-openapi-typescript.ts (hunk 1 above)
Comment (HeverFarber) [reply src/serverless-openapi-typescript.ts:5]: why? (But I didn't do it - I just convert old js file to TS file)

File: src/serverless-openapi-typescript.ts (hunk 4 above)
Comment (HeverFarber) [reply src/serverless-openapi-typescript.ts]: ee6a838 for now

File: .gitignore (hunk 2 above)
Comment (HeverFarber) [reply .gitignore:6]: @roni-frantchi @RLRabinowitz

File: src/serverless-openapi-typescript.ts (hunk 4 above)
Comment (roni-frantchi) [reply src/serverless-openapi-typescript.ts]: > I thought we don't like it  

Why not? It's a great TS feature.

File: .gitignore (hunk 2 above)
Comment (HeverFarber) [reply .gitignore:6]: After we discuss. I will return it back For CI

File: src/serverless-openapi-typescript.ts (hunk 4 above)
Comment (HeverFarber) [reply src/serverless-openapi-typescript.ts]: I agree. I love it

File: .gitignore (hunk 2 above)
Comment (HeverFarber) [reply .gitignore]: f4b0d52

**Review by roni-frantchi — APPROVED** [review]

**Pull request merged**